WL-3797 Changed Label of Joinable Set to Joinable Group.

ouit0408 opened the pull request:
"Delete all groups" button shifted to right on 'Edit Joinable Group' screen.
Confirm message comes when user clicks "Delete all groups" button and for
that have created 'JoinableGroupDelProducer.java',
'JoinableGroupDelViewParameters.java' and 'JoinableGroupDel.html'.Also, have
edited 'CreateJoinableGroups.html' and 'CreateJoinableGroupsProducer.java'.

ouit0408 committed:
WL-3797 Changed Label of Joinable Set to Joinable Group.

"Delete all groups" button shifted to right on 'Edit Joinable Group' screen.
Confirm message comes when user clicks "Delete all groups" button and for
that have created 'JoinableGroupDelProducer.java',
'JoinableGroupDelViewParameters.java' and 'JoinableGroupDel.html'.Also, have
edited 'CreateJoinableGroups.html' and 'CreateJoinableGroupsProducer.java'.

buckett commented:
I think this pull request contains more than just the changes to the label. It looks like it contains the changes to confirm the joinable set deletion as well.

ouit0408 commented:
Yes it does contain those changes, have added details in my comments

ouit0408 commented:
Yes it does contain those changes, have added details in my comments.by mistake i clicked 'close and comment', sorry, opened it again.

buckett commented on site-manage-group-section-role-helper/tool/src/java/org/sakaiproject/site/tool/helper/managegroupsectionrole/rsf/JoinableGroupDelProducer.java:
```diff
@@ -0,0 +1,98 @@
+ * @author
+ *
+ */
+public class JoinableGroupDelProducer
```
The indentation on the file is a mixture of tabs and spaces.

buckett commented on site-manage-group-section-role-helper/tool/src/java/org/sakaiproject/site/tool/helper/managegroupsectionrole/rsf/JoinableGroupDelViewParameters.java:
```diff
@@ -0,0 +1,30 @@
+
+import uk.org.ponder.rsf.viewstate.SimpleViewParameters;
+
+public class JoinableGroupDelViewParameters extends SimpleViewParameters{
```
Indentation and formatting is inconsistent.

ouit0408 replied:
have added indentation 'to tabs' in both the files.

buckett replied:
I can't see these changes on github.

ouit0408 committed:
WL-3797 Change label of Joinable set to Joinable Group and add deletion confirmation

changed indentation 'to tabs' in files 'JoinableGroupDelViewParameters.java'
and 'JoinableGroupDelProducer.java'

buckett commented on site-manage-group-section-role-helper/tool/src/java/org/sakaiproject/site/tool/helper/managegroupsectionrole/rsf/CreateJoinableGroupsProducer.java:
```diff
@@ -122,7 +114,9 @@ public void fillComponents(UIContainer tofill, ViewParameters viewparams,
         
         if(edit){
         	//Delete Set button:
-        	UICommand.make(groupForm, "delete", messageLocator.getMessage("group.joinable.delete"), "#{SiteManageGroupSectionRoleHandler.processDeleteJoinableSet}");
```
Indentation.

buckett commented on site-manage-group-section-role-helper/tool/src/java/org/sakaiproject/site/tool/helper/managegroupsectionrole/rsf/JoinableGroupDelProducer.java:
```diff
@@ -0,0 +1,98 @@
+	public static final String VIEW_ID = "JoinableGroupDel";
+	public MessageLocator messageLocator;
+	public SiteManageGroupSectionRoleHandler handler;
+	public FrameAdjustingProducer frameAdjustingProducer;
```
You don't seem to be using this. If you want the frame to automagically adjust you need to added it to the components.

buckett commented on site-manage-group-section-role-helper/tool/src/java/org/sakaiproject/site/tool/helper/managegroupsectionrole/rsf/JoinableGroupDelProducer.java:
```diff
@@ -0,0 +1,98 @@
+
+		String joinableSetId = ((JoinableGroupDelViewParameters) viewparams).id;
+		boolean delete = joinableSetId != null && !"".equals(joinableSetId);
+		if(delete){
```
Why would delete ever be false?

buckett commented on site-manage-group-section-role-helper/tool/src/java/org/sakaiproject/site/tool/helper/managegroupsectionrole/rsf/JoinableGroupDelProducer.java:
```diff
@@ -0,0 +1,98 @@
+				}
+			}
+
+		}
```
If you expect it not to be in delete mode it might be nice to have an error message saying that the joinable group wasn't supplied.

buckett commented on site-manage-group-section-role-helper/tool/src/java/org/sakaiproject/site/tool/helper/managegroupsectionrole/rsf/JoinableGroupDelProducer.java:
```diff
@@ -0,0 +1,98 @@
+		String joinableSetId = ((JoinableGroupDelViewParameters) viewparams).id;
+		boolean delete = joinableSetId != null && !"".equals(joinableSetId);
+		if(delete){
+			handler.joinableSetName = joinableSetId;
```
Why do you need to set the handler up here as it's a request scoped bean so will be thrown away after this request and the change lost.

ouit0408 committed:
WL-3797 Corrected the indentation.
Removed components which were not used from JoinableGroupDelProducer.java.
Added error message saying that the joinable group wasn't supplied if 'delete'
mode not supplied.
